owncloudApp/src/main/java/com/owncloud/android/authentication/PatternManager.java

```diff
@@ -99,28 +109,23 @@ private void checkPattern(Activity activity) {
         activity.startActivity(i);
     }
 
-    private boolean patternShouldBeRequested() {
-        int PATTERN_TIMEOUT = 1000;
-        if ((SystemClock.elapsedRealtime() - timeStamp) > PATTERN_TIMEOUT &&
-                mVisibleActivitiesCounter <= 0
-        ) {
-            return isPatternEnabled();
-        }
-        return false;
-    }
-
-    public boolean isPatternEnabled() {
-        SharedPreferences appPrefs = PreferenceManager.getDefaultSharedPreferences(MainApp.Companion.getAppContext());
-        return appPrefs.getBoolean(PatternLockActivity.PREFERENCE_SET_PATTERN, false);
+    public void onBiometricCancelled(Activity activity) {
+        // Ask user for pattern
+        checkPattern(activity);
     }
 
     /**
      * This can be used for example for onActivityResult, where you don't want to re authenticate
      * again.
-     * <p>
+     *
      * USE WITH CARE
      */
     public void bayPassUnlockOnce() {
-        setUnlockTimestamp();
+        int timeout = LockTimeout.Companion.fromStringToMilliseconds(preferencesProvider.getString(LOCK_TIMEOUT, LockTimeout.IMMEDIATELY.name()));
+        long lastUnlockTimestamp = preferencesProvider.getLong(LAST_UNLOCK_TIMESTAMP, 0);
+        if (System.currentTimeMillis() - lastUnlockTimestamp > timeout) {
```

Comment:
I'm pretty sure I have seen this several times 😄  Check the previous review comment 😉